File: finance-api/src/main/java/com/financesystem/finance_api/modules/tenant/fx/domain/model/FxExchangeRate.java
package com.financesystem.finance_api.modules.tenant.fx.domain.model;

import com.financesystem.finance_api.modules.tenant.accounts.domain.model.CurrencyCode;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.UUID;

public record FxExchangeRate(
        UUID id,
        CurrencyCode sourceCurrency,
        CurrencyCode targetCurrency,
        BigDecimal rate,
        boolean active,
        String description,
        Instant createdAt,
        Instant updatedAt
) {
}

File: finance_mobile/lib/domain/entities/fx_exchange_rate.dart
class FxExchangeRate {
  final String id;
  final String sourceCurrency;
  final String targetCurrency;
  final double rate;
  final bool active;
  final String? description;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  const FxExchangeRate({
    required this.id,
    required this.sourceCurrency,
    required this.targetCurrency,
    required this.rate,
    required this.active,
    required this.description,
    required this.createdAt,
    required this.updatedAt,
  });
}

class FxExchangeRatesPage {
  final List<FxExchangeRate> items;
  final int totalPages;
  final int totalElements;
  final int number;
  final int size;
  final bool first;
  final bool last;

  const FxExchangeRatesPage({
    required this.items,
    required this.totalPages,
    required this.totalElements,
    required this.number,
    required this.size,
    required this.first,
    required this.last,
  });
}
